platform/mellanox/docker-syncd-mlnx/ecmp_calc_sdk.py

```diff
@@ -0,0 +1,466 @@
+            if next_hop.next_hop_key.type == SX_NEXT_HOP_TYPE_IP:
+                next_hops.append(next_hop)
+                
+        return next_hops
+    
+    finally:
+        delete_uint32_t_p(next_hop_count_p)
+        delete_sx_next_hop_t_arr(next_hop_list_p)
+
+def sx_get_router_interface(handle, vrid, rif):
+    """ Get router interface information
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        vrid (sx_router_id_t): virtual router id
+        rif (sx_router_interface_t): router interface id
+    
+    Returns:
+        dict : Dictionary contains interface parameters
+    """       
+    try:
+        vrid_p = new_sx_router_id_t_p()
+        sx_router_id_t_p_assign(vrid_p, vrid)
+    
+        ifc_p = new_sx_router_interface_param_t_p()
+        ifc_attr_p = new_sx_interface_attributes_t_p()
+        rif_params = {}
+        
+        rc = sx_api_router_interface_get(handle, rif, vrid_p, ifc_p, ifc_attr_p)
+        sx_check_rc(rc)
+        
+        if ifc_p.type == SX_L2_INTERFACE_TYPE_PORT_VLAN:
+            rif_params['port'] = ifc_p.ifc.port_vlan.port
+            
+        if ifc_p.type == SX_L2_INTERFACE_TYPE_VPORT:
+            rif_params['vport'] = ifc_p.ifc.vport.vport        
+    
+        if ifc_p.type == SX_L2_INTERFACE_TYPE_VLAN:
+            rif_params['vlan'] = ifc_p.ifc.vlan.vlan
+            
+        return rif_params
+        
+    finally:
+        delete_sx_router_id_t_p(vrid_p)
+        delete_sx_router_interface_param_t_p(ifc_p)
+        delete_sx_interface_attributes_t_p(ifc_attr_p)    
+
+def sx_port_vport_base_get(handle, vport):
+    """ Get SDK logical index for given vport
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        vport (sx_port_id_t): SDK vport id
+
+    Returns:
+        sx_port_log_id_t : SDK logical index 
+    """      
+    try:
+        vlan_id_p = new_sx_vlan_id_t_p()
+        logical_port_p = new_sx_port_log_id_t_p()
+        
+        rc = sx_api_port_vport_base_get(handle, vport, vlan_id_p, logical_port_p)
+        sx_check_rc(rc)
+        
+        logical_port = sx_port_log_id_t_p_value(logical_port_p)
+        
+        return logical_port  
+      
+    finally:
+        delete_sx_vlan_id_t_p(vlan_id_p)
+        delete_sx_port_log_id_t_p(logical_port_p)   
+        
+def sx_router_neigh_get_mac(handle, rif, addr, type_ipv6=False):
+    """ Get neighbour mac address
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        rif (sx_port_id_t): SDK vport id
+        addr (sx_ip_addr_t): Neighbour IP address
+        type_ipv6 (bool): True for IPv6
+
+    Returns:
+        str : Neighbour mac address
+    """     
+    try:
+        neigh_entry_cnt_p = new_uint32_t_p()
+        neigh_entry_list_p = new_sx_neigh_get_entry_t_arr(1)
+
+        filter_p = new_sx_neigh_filter_t_p()
+        neigh_filter = sx_neigh_filter_t()
+        neigh_filter.filter_by_rif = SX_KEY_FILTER_FIELD_NOT_VALID
+        neigh_filter.rif = 0
```

Comment:
@liorghub please avoid magic numbers

Reply:
It is not a magic number, it is an initialization. I think that define here will an overhead, let me know if you think otherwise.